src/renderer/renderer.cpp

```diff
@@ -1,26 +1,37 @@
+#include "plugins/plugin_interface.h"
 #include "renderer/renderer_callback_registry.h"
-#include "renderer/renderer_plugin_import.h"
 
 #include <nox/renderer.h>
 
+#if defined(NOX_BUILD_RENDERER_OPENGL)
+NOX_PLUGIN_LOAD(OpenGL);
+#endif
+
 namespace NOX {
 
-RendererPtr RendererFactory::createRenderer(RendererBackend backend) {
+namespace {
+
+bool isRendererPluginRegistered(RendererBackend backend) {
     NOX_ASSERT(backend == RendererBackend::NONE);
 
-    if (!RendererCallbackRegistry::isCallbackRegistered(backend)) {
-        bool result = false;
-        if (backend == RendererBackend::OPENGL) {
-            result = NOX_PLUGIN_LOAD_OPENGL();
-        }
+    if (backend == RendererBackend::OPENGL) {
+        return OpenGLPlugin::pluginRegistered;
+    }
+
+    return false;
+}
+
+} // namespace
+
+RendererPtr RendererFactory::createRenderer(RendererBackend backend) {
+    NOX_ASSERT(backend == RendererBackend::NONE);
 
-        if (!result) {
-            NOX_ASSERT(true);
-            return nullptr;
-        }
+    if (!isRendererPluginRegistered(backend)) {
+        NOX_ASSERT(true);
+        return nullptr;
     }
 
-    auto callback = RendererCallbackRegistry::getCallback(backend);
+    auto callback = RendererCallbackRegistry::instance().getCallback(backend);
```

Comment:
structured bindings can be used here
auto [createRenderer,destroyRenderer] = RendererCallbackRegistry::instance().getCallback(backend);

Reply:
Good point. Done.